index.js

```diff
@@ -428,6 +428,8 @@ function TChannelConnection(channel, socket, direction, remoteAddr) {
 
     self.handler.on('error', function onHandlerError(err) {
         self.resetAll(err);
+        // resetAll() does not close the socket
```

Comment:
It cant because the server path calls `socket.end(); resetAll()`. We cannot call destroy in the `resetAll()` method because its used for both clean exit and shutdown.

It would also be weird to call `socket.destroy()` in the close event listener on the socket.

The fact that `resetAll()` is used for both "destroy" and "clean close of socket" is what makes it difficult.